### DISCUSSION TITLE
When -trim does not trim

### DISCUSSION THREAD

**Discussion by John-Whitehead** [ImageMagick/ImageMagick#6298] [Digital Image Processing]
Hello All - I am starting a project to digitize a bunch of old family photos.

I use Fedora 38 and have an old HP scanner.  Imagemagick at  7.1.1-4

To scan a photo, I used this command line tool:
`scanimage --device-name="genesys:libusb:001:002" --format=png --mode=color --depth=16 --resolution=200 --output=image.png`

This created a PNG file,  1712x2480 with the photo in the top left corner and a ton of whitespace as the scanner scanned the entire bed.

I wanted to use Imagemagick to get rid of the white space and did this simple command:

`magick image.png -trim image_cropped.png`

and I got back the same image.  I tried +repage and played with -fuzz but always get the same result.  Am I missing something?  These photos are of varying shapes and sizes and I don't want to manually crop them.

Thanks!

**Comment by fmw42** [comment]
You should add -fuzz XX% before the trim and also +repage after the trim. However, you may have small black specks that are preventing the trim.  Please post an example of your scan.

**Comment by John-Whitehead** [comment]
I used a different image so the file was smaller.  Created it with this command:
`scanimage --device-name="genesys:libusb:001:002" --format=png --mode=color --depth=8 --resolution=200 -y 150 --output=image.png`

Output file is attached, I could not trim that one either.

![image_cropped](https://user-images.githubusercontent.com/40612016/236572564-a5e8e98e-4280-4965-babf-b4fe45894755.png)

**Reply by fmw42** [reply]
There is nothing that can be trimmed.  You need to have solid color (or nearly so) rows or columns for trim to work in those areas. Also your large gray-blue color at the bottom not only has a black stripe on the left, but has small dark specks within it.

**Comment by fmw42** [comment]
Try this on IM 7

`magick img.png -background "#ccc8ec" -define trim:percent-background=0% -fuzz 2% -trim +repage -trim +repage result.png`

![result](https://user-images.githubusercontent.com/13458218/236578228-22f1985e-86b6-4ff1-aa76-d52b81d60832.png)


However, the background color and fuzz amount may be image dependent.
